Compile warnings with gcc

Version: https://github.com/azerothcore/azerothcore-wotlk/commit/db25b814e2587f51e3ec8a62186012c2380aaef3

Using compiler: `gcc version 7.3.0 (Ubuntu 7.3.0-27ubuntu1~18.04) `

Currently there are the following warnings:

> [ 34%] Building CXX object src/server/game/CMakeFiles/game.dir/Handlers/AuctionHouseHandler.cpp.o
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp: In member function ‘void Guild::HandleSetBankTabInfo(WorldSession*, uint8, const string&, const string&)’:
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp:1411:6: warning: ‘%u’ directive writing between 1 and 3 bytes into a region of size 2 [-Wformat-overflow=]
>  void Guild::HandleSetBankTabInfo(WorldSession* session, uint8 tabId, std::string const& name, std::string const& icon)
>       ^~~~~
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp:1411:6: note: directive argument in the range [0, 255]
> In file included from /usr/include/stdio.h:862:0,
>                  from /usr/include/ace/os_include/os_stdio.h:30,
>                  from /usr/include/ace/os_include/os_limits.h:26,
>                  from /usr/include/ace/Basic_Types.h:52,
>                  from /home/shin/sources/azerothcore-wotlk/src/server/game/Server/WorldSocket.h:16,
>                  from PrecompiledHeaders/gamePCH.h:3:
> /usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 2 and 4 bytes into a destination of size 2
>        __bos (__s), __fmt, __va_arg_pack ());
>                                            ^
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp: In member function ‘void Guild::HandleSetRankInfo(WorldSession*, uint8, const string&, uint32, uint32, GuildBankRightsAndSlotsVec)’:
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp:1444:6: warning: ‘%u’ directive writing between 1 and 3 bytes into a region of size 2 [-Wformat-overflow=]
>  void Guild::HandleSetRankInfo(WorldSession* session, uint8 rankId, std::string const& name, uint32 rights, uint32 moneyPerDay, GuildBankRightsAndSlotsVec rightsAndSlots)
>       ^~~~~
> /home/shin/sources/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp:1444:6: note: directive argument in the range [0, 255]
> In file included from /usr/include/stdio.h:862:0,
>                  from /usr/include/ace/os_include/os_stdio.h:30,
>                  from /usr/include/ace/os_include/os_limits.h:26,
>                  from /usr/include/ace/Basic_Types.h:52,
>                  from /home/shin/sources/azerothcore-wotlk/src/server/game/Server/WorldSocket.h:16,
>                  from PrecompiledHeaders/gamePCH.h:3:
> /usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 2 and 4 bytes into a destination of size 2
>        __bos (__s), __fmt, __va_arg_pack ());
>                                            ^
>

--- a/src/server/game/Guilds/Guild.cpp
+++ b/src/server/game/Guilds/Guild.cpp
@@ -1418,11 +1418,8 @@ void Guild::HandleSetBankTabInfo(WorldSession* session, uint8 tabId, std::string
         return;
     }
 
-    char aux[2];
-    sprintf(aux, "%u", tabId);
-
     tab->SetInfo(name, icon);
-    _BroadcastEvent(GE_BANK_TAB_UPDATED, 0, aux, name.c_str(), icon.c_str());
+    _BroadcastEvent(GE_BANK_TAB_UPDATED, 0, std::to_string(tabId).c_str(), name.c_str(), icon.c_str());
 }
 
 void Guild::HandleSetMemberNote(WorldSession* session, std::string const& name, std::string const& note, bool isPublic)
@@ -1459,9 +1456,7 @@ void Guild::HandleSetRankInfo(WorldSession* session, uint8 rankId, std::string c
         for (GuildBankRightsAndSlotsVec::const_iterator itr = rightsAndSlots.begin(); itr != rightsAndSlots.end(); ++itr)
             _SetRankBankTabRightsAndSlots(rankId, *itr);
 
-        char aux[2];
-        sprintf(aux, "%u", rankId);
-        _BroadcastEvent(GE_RANK_UPDATED, 0, aux, name.c_str());
+        _BroadcastEvent(GE_RANK_UPDATED, 0, std::to_string(rankId).c_str(), name.c_str());
     }
 }
 
@@ -1686,11 +1681,7 @@ void Guild::HandleAddNewRank(WorldSession* session, std::string const& name)
     // Only leader can add new rank
     if (_IsLeader(session->GetPlayer()))
         if (_CreateRank(name, GR_RIGHT_GCHATLISTEN | GR_RIGHT_GCHATSPEAK))
-        {
-            char aux[2];
-            sprintf(aux, "%u", size);
-            _BroadcastEvent(GE_RANK_UPDATED, 0, aux, name.c_str());
-        }
+            _BroadcastEvent(GE_RANK_UPDATED, 0, std::to_string(size).c_str(), name.c_str());
 }
 
 void Guild::HandleRemoveLowestRank(WorldSession* session)

